From cf7363b42b2fdc9fd108bed8d53b35adf4d52ad5 Mon Sep 17 00:00:00 2001
From: H.J. Lu <hjl.tools@gmail.com>
Date: Mon, 29 Sep 2014 08:35:49 -0700
Subject: [PATCH] Fix build for OLD_FREEBSD_ABI_LABEL

	PR ld/17440
	* elf32-i386.c (elf_i386_fbsd_post_process_headers): Fix build
	for OLD_FREEBSD_ABI_LABEL.
---
 bfd/ChangeLog    |    6 ++++++
 bfd/elf32-i386.c |    7 +++++--
 2 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/bfd/ChangeLog b/bfd/ChangeLog
index f70cad6..e4445dc 100644
--- a/bfd/ChangeLog
+++ b/bfd/ChangeLog
@@ -1,3 +1,9 @@
+2014-09-29  H.J. Lu  <hongjiu.lu@intel.com>
+
+	PR ld/17440
+	* elf32-i386.c (elf_i386_fbsd_post_process_headers): Fix build
+	for OLD_FREEBSD_ABI_LABEL.
+
 2014-09-24  Markus Trippelsdorf  <markus@trippelsdorf.de>
 
 	PR 17422
diff --git a/bfd/elf32-i386.c b/bfd/elf32-i386.c
index a00d47c..afa21b5 100644
--- a/bfd/elf32-i386.c
+++ b/bfd/elf32-i386.c
@@ -5120,8 +5120,11 @@ elf_i386_fbsd_post_process_headers (bfd *abfd, struct bfd_link_info *info)
   _bfd_elf_post_process_headers (abfd, info);
 
 #ifdef OLD_FREEBSD_ABI_LABEL
-  /* The ABI label supported by FreeBSD <= 4.0 is quite nonstandard.  */
-  memcpy (&i_ehdrp->e_ident[EI_ABIVERSION], "FreeBSD", 8);
+  {
+    /* The ABI label supported by FreeBSD <= 4.0 is quite nonstandard.  */
+    Elf_Internal_Ehdr *i_ehdrp = elf_elfheader (abfd);
+    memcpy (&i_ehdrp->e_ident[EI_ABIVERSION], "FreeBSD", 8);
+  }
 #endif
 }
 
-- 
1.7.1

